flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/PlannerBase.scala

```diff
@@ -112,24 +114,26 @@ abstract class PlannerBase(
     val planner = getFlinkPlanner
     // parse the sql query
     val parsed = planner.parse(stmt)
-    if (null != parsed && parsed.getKind.belongsTo(SqlKind.QUERY)) {
-      // validate the sql query
-      val validated = planner.validate(parsed)
-      // transform to a relational tree
-      val relational = planner.rel(validated)
-      List(new PlannerQueryOperation(relational.project()))
-    } else if (null != parsed && parsed.isInstanceOf[SqlInsert]) {
-      val insert = parsed.asInstanceOf[SqlInsert]
-      // validate the SQL query
-      val query = insert.getSource
-      val validatedQuery = planner.validate(query)
-      val sinkOperation = new CatalogSinkModifyOperation(
-        insert.getTargetTable.asInstanceOf[SqlIdentifier].names,
-        new PlannerQueryOperation(planner.rel(validatedQuery).rel)
-      )
-      List(sinkOperation)
-    } else {
-      throw new TableException(s"Unsupported query: $stmt")
+    parsed match {
+      case insert: SqlInsert =>
+        // get name of sink table
+        val targetTablePath = insert.getTargetTable.asInstanceOf[SqlIdentifier].names
+
+        List(new CatalogSinkModifyOperation(targetTablePath,
+          SqlToOperationConverter.convert(planner,
+            insert.getSource).asInstanceOf[PlannerQueryOperation])
+          .asInstanceOf[Operation]).asJava
```

Comment:
asJava is unnecessary?

Reply:
Yes, cause we already have `JavaConversions._`